data/src/main/java/me/androidbox/data/alarm_manager/AlarmSchedulerImp.kt

```diff
@@ -0,0 +1,45 @@
+package me.androidbox.data.alarm_manager
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import dagger.hilt.android.qualifiers.ApplicationContext
+import me.androidbox.domain.alarm_manager.AlarmItem
+import me.androidbox.domain.alarm_manager.AlarmScheduler
+import javax.inject.Inject
+
+class AlarmSchedulerImp @Inject constructor(
+    @ApplicationContext private val context: Context,
+) : AlarmScheduler {
+
+    private val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
+
+    override fun schedule(item: AlarmItem) {
+        val intent = Intent(context, AlarmReceiver::class.java).apply {
+            putExtra("EXTRA_MESSAGE", item.message)
+        }
+
+        alarmManager.setExact(
+            AlarmManager.RTC_WAKEUP,
+            item.dateTime.toEpochSecond() * 1000,
```

Comment:
As I am using API 21 I need the AndroidThreeTen library to be able to do this `toEpochSeconds()`

Reply:
I don't understand why a library is needed to convert a `LocalDateTime` to millis? 🤔 
This will do it:
```
dateTime.atZone(ZoneId.systemDefault()).toEpochSecond() * 1000
```

Reply:
@philipplackner Removed the ThreeTen library. The reason I was having trouble was because I forgot to add the `coreLibraryDesugaring` in my data layer module